targets/tests/test_topics.py

```diff
@@ -0,0 +1,46 @@
+from django.core.files.storage import default_storage
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from targets.tests.factories import TopicFactory
+from users.tests.factories import UserFactory
+
+
+class TopicTests(APITestCase):
+    AUTHENTICATION_ERROR_MESSAGE = 'Authentication credentials were not provided.'
+
+    @classmethod
+    def setUpClass(self):
+        self.topic1 = TopicFactory()
+        self.topic1.save()
+        self.topic2 = TopicFactory()
+        self.topic2.save()
+        self.url = reverse('topics-list')
+        self.user = UserFactory()
+        self.user.save()
+
+    def test_list_topics_logged_in(self):
+        self.client.force_authenticate(self.user)
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(len(response.data), 2)
+        self.assertEqual(response.data[0]['title'], self.topic1.title)
+        self.assertEqual(response.data[0]['image'].split('?')[0], self.topic1.image.url.split('?')[0])
+        self.assertEqual(response.data[1]['title'], self.topic2.title)
+        self.assertEqual(response.data[1]['image'].split('?')[0], self.topic2.image.url.split('?')[0])
+
+    def test_list_topics_not_logged_in(self):
+        response = self.client.get(self.url)
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+        self.assertTrue(response.data['detail'] == TopicTests.AUTHENTICATION_ERROR_MESSAGE)
+
+    def test_topic_image_name(self):
+        self.assertEqual(self.topic1.image.name, self.topic1.title)
+
+    @classmethod
+    def tearDownClass(self):
+        default_storage.delete(self.topic1.title)
+        default_storage.delete(self.topic2.title)
```

Comment:
When you use `tearDownClass` or `setUpClass` is a good practice to call `super` function.
In the case of `tearDownClass` call it at the end, and in `setUpClass` call it at the beginning, take a look at:
https://docs.djangoproject.com/en/3.1/topics/testing/tools/#django.test.SimpleTestCase.databases